diff --git a/dlls/nvencodeapi/nvencodeapi.c b/dlls/nvencodeapi/nvencodeapi.c
--- a/dlls/nvencodeapi/nvencodeapi.c
+++ b/dlls/nvencodeapi/nvencodeapi.c
@@ -35,6 +35,8 @@ static LINUX_NV_ENCODE_API_FUNCTION_LIST origFunctions;
 static NVENCSTATUS (*pNvEncodeAPICreateInstance)(LINUX_NV_ENCODE_API_FUNCTION_LIST *functionList);
 static NVENCSTATUS (*pNvEncodeAPIGetMaxSupportedVersion)(uint32_t* version);
 
+#include "uu-remote-nvenc-bridge.h"
+
 static NVENCSTATUS WINAPI NvEncOpenEncodeSession(void *device, uint32_t deviceType, void **encoder)
 {
     TRACE("(%p, %u, %p)\n", device, deviceType, encoder);
@@ -77,7 +79,7 @@ static NVENCSTATUS WINAPI NvEncGetInputFormats(void *encoder, GUID encodeGUID, NV_ENC_BUFFER_FORMAT *inputFmts,
                                                uint32_t inputFmtArraySize, uint32_t *inputFmtCount)
 {
     TRACE("(%p, %s, %p)\n", encoder, debugstr_guid(&encodeGUID), inputFmtCount);
-    return origFunctions.nvEncGetInputFormatCount(encoder, encodeGUID, inputFmtCount);
+    return origFunctions.nvEncGetInputFormats(encoder, encodeGUID, inputFmts, inputFmtArraySize, inputFmtCount);
 }
 
 static NVENCSTATUS WINAPI NvEncGetEncodeCaps(void *encoder, GUID encodeGUID, NV_ENC_CAPS_PARAM *capsParam, int *capsVal)
@@ -125,7 +127,7 @@ static NVENCSTATUS WINAPI NvEncInitializeEncoder(void *encoder, NV_ENC_INITIALIZE_PARAMS *createEncodeParams)
         linux_encode_params.enableEncodeAsync = 0;
     }
 
-    return origFunctions.nvEncInitializeEncoder(encoder, createEncodeParams);
+    return uu_initialize_encoder(encoder, createEncodeParams);
 }
 
 static NVENCSTATUS WINAPI NvEncCreateInputBuffer(void *encoder, NV_ENC_CREATE_INPUT_BUFFER *createInputBufferParams)
@@ -143,6 +145,6 @@ static NVENCSTATUS WINAPI NvEncCreateBitstreamBuffer(void *encoder, NV_ENC_CREATE_BITSTREAM_BUFFER *createBitstreamBufferParams)
 {
     TRACE("(%p, %p)\n", encoder, createBitstreamBufferParams);
-    return origFunctions.nvEncCreateBitstreamBuffer(encoder, createBitstreamBufferParams);
+    return uu_create_bitstream_buffer(encoder, createBitstreamBufferParams);
 }
 
 static NVENCSTATUS WINAPI NvEncDestroyBitstreamBuffer(void *encoder, NV_ENC_OUTPUT_PTR bitstreamBuffer)
@@ -149,6 +151,6 @@ static NVENCSTATUS WINAPI NvEncDestroyBitstreamBuffer(void *encoder, NV_ENC_OUTPUT_PTR bitstreamBuffer)
 {
     TRACE("(%p, %p)\n", encoder, bitstreamBuffer);
-    return origFunctions.nvEncDestroyBitstreamBuffer(encoder, bitstreamBuffer);
+    return uu_destroy_bitstream_buffer(encoder, bitstreamBuffer);
 }
 
 static NVENCSTATUS WINAPI NvEncEncodePicture(void *encoder, NV_ENC_PIC_PARAMS *encodePicParams)
@@ -157,7 +159,7 @@ static NVENCSTATUS WINAPI NvEncEncodePicture(void *encoder, NV_ENC_PIC_PARAMS *encodePicParams)
     NVENCSTATUS result;
     TRACE("(%p, %p)\n", encoder, encodePicParams);
 
-    result = origFunctions.nvEncEncodePicture(encoder, encodePicParams);
+    result = uu_encode_picture(encoder, encodePicParams);
 
     if (encodePicParams->completionEvent)
         SetEvent(encodePicParams->completionEvent);
@@ -168,6 +170,6 @@ static NVENCSTATUS WINAPI NvEncLockBitstream(void *encoder, NV_ENC_LOCK_BITSTREAM *lockBitstreamBufferParams)
 {
     TRACE("(%p, %p)\n", encoder, lockBitstreamBufferParams);
-    return origFunctions.nvEncLockBitstream(encoder, lockBitstreamBufferParams);
+    return uu_lock_bitstream(encoder, lockBitstreamBufferParams);
 }
 
 static NVENCSTATUS WINAPI NvEncUnlockBitstream(void *encoder, NV_ENC_OUTPUT_PTR bitstreamBuffer)
@@ -174,6 +176,6 @@ static NVENCSTATUS WINAPI NvEncUnlockBitstream(void *encoder, NV_ENC_OUTPUT_PTR bitstreamBuffer)
 {
     TRACE("(%p, %p)\n", encoder, bitstreamBuffer);
-    return origFunctions.nvEncUnlockBitstream(encoder, bitstreamBuffer);
+    return uu_unlock_bitstream(encoder, bitstreamBuffer);
 }
 
 static NVENCSTATUS WINAPI NvEncLockInputBuffer(void *encoder, NV_ENC_LOCK_INPUT_BUFFER *lockInputBufferParams)
@@ -220,12 +222,12 @@ static NVENCSTATUS WINAPI NvEncMapInputResource(void *encoder, NV_ENC_MAP_INPUT_RESOURCE *mapInputResParams)
 {
     TRACE("(%p, %p)\n", encoder, mapInputResParams);
-    return origFunctions.nvEncMapInputResource(encoder, mapInputResParams);
+    return uu_upload_and_map_resource(encoder, mapInputResParams);
 }
 
 static NVENCSTATUS WINAPI NvEncUnmapInputResource(void *encoder, NV_ENC_INPUT_PTR mappedInputBuffer)
 {
     TRACE("(%p, %p)\n", encoder, mappedInputBuffer);
-    return origFunctions.nvEncUnmapInputResource(encoder, mappedInputBuffer);
+    return uu_unmap_input_resource(encoder, mappedInputBuffer);
 }
 
 static NVENCSTATUS WINAPI NvEncDestroyEncoder(void *encoder)
@@ -232,6 +234,6 @@ static NVENCSTATUS WINAPI NvEncDestroyEncoder(void *encoder)
 {
     TRACE("(%p)\n", encoder);
-    return origFunctions.nvEncDestroyEncoder(encoder);
+    return uu_destroy_encoder(encoder);
 }
 
 static NVENCSTATUS WINAPI NvEncInvalidateRefFrames(void *encoder, uint64_t invalidRefFrameTimeStamp)
@@ -244,24 +246,24 @@ static NVENCSTATUS WINAPI NvEncOpenEncodeSessionEx(NV_ENC_OPEN_ENCODE_SESSION_EX_PARAMS *openSessionExParams, void **encoder)
 {
     TRACE("(%p, %p)\n", openSessionExParams, encoder);
-    return origFunctions.nvEncOpenEncodeSessionEx(openSessionExParams, encoder);
+    return uu_open_d3d11_session(openSessionExParams, encoder);
 }
 
 static NVENCSTATUS WINAPI NvEncRegisterResource(void *encoder, NV_ENC_REGISTER_RESOURCE *registerResParams)
 {
     TRACE("(%p, %p)\n", encoder, registerResParams);
-    return origFunctions.nvEncRegisterResource(encoder, registerResParams);
+    return uu_register_d3d11_resource(encoder, registerResParams);
 }
 
 static NVENCSTATUS WINAPI NvEncUnregisterResource(void *encoder, NV_ENC_REGISTERED_PTR registeredResource)
 {
     TRACE("(%p, %p)\n", encoder, registeredResource);
-    return origFunctions.nvEncUnregisterResource(encoder, registeredResource);
+    return uu_unregister_resource(encoder, registeredResource);
 }
 
 static NVENCSTATUS WINAPI NvEncReconfigureEncoder(void *encoder, NV_ENC_RECONFIGURE_PARAMS *reInitEncodeParams)
 {
     TRACE("(%p, %p)\n", encoder, reInitEncodeParams);
-    return origFunctions.nvEncReconfigureEncoder(encoder, reInitEncodeParams);
+    return uu_reconfigure_encoder(encoder, reInitEncodeParams);
 }
 
 static NVENCSTATUS WINAPI NvEncCreateMVBuffer(void *encoder, NV_ENC_CREATE_MV_BUFFER *createMVBufferParams)
